src/Slices/DesiredStateCompare/UI/Page.tsx

```diff
@@ -13,65 +13,70 @@ export const Page: React.FC = () => {
 };
 
 export const View: React.FC<Diff.Identifiers> = ({ from, to }) => {
-  const { queryResolver } = useContext(DependencyContext);
   const refs: DiffWizard.Refs = useRef({});
   const [statuses, setStatuses] = useState(Diff.defaultStatuses);
   const [searchFilter, setSearchFilter] = useState("");
 
-  const [data] = queryResolver.useOneTime<"GetDesiredStateDiff">({
-    kind: "GetDesiredStateDiff",
+  const { data, isError, error, isSuccess, refetch } = useGetDesiredStateDiff().useOneTime(
     from,
-    to,
-  });
-
-  const filteredData = RemoteData.mapSuccess(
-    (resources) =>
-      resources
-        .filter((resource) => statuses.includes(resource.status))
-        .filter((resource) =>
-          resource.resource_id.toLocaleLowerCase().includes(searchFilter.toLocaleLowerCase())
-        ),
-    data
+    to
   );
 
-  return (
-    <>
-      <PageSection>
-        <Content>
-          <Content component="h1">{words("desiredState.compare.title")}</Content>
-        </Content>
-      </PageSection>
-      <PageSection hasBodyWrapper={false}>
-        <Toolbar>
-          <ToolbarContent style={{ padding: 0 }}>
-            <DiffWizard.DiffPageFilter
-              statuses={statuses}
-              setStatuses={setStatuses}
-              searchFilter={searchFilter}
-              setSearchFilter={setSearchFilter}
-            />
-          </ToolbarContent>
-        </Toolbar>
-      </PageSection>
-      <PageSection hasBodyWrapper={false} hasShadowBottom>
-        <DiffWizard.Controls data={filteredData} refs={refs} from={from} to={to} />
-      </PageSection>
-      <PageSection hasBodyWrapper={false} isFilled>
-        <RemoteDataView
-          data={filteredData}
-          label="CompareView"
-          SuccessView={(resources) =>
-            resources.length <= 0 ? (
-              <EmptyView message={words("desiredState.compare.empty")} />
-            ) : (
-              <DiffWizard.ItemList
-                items={resources.map(DiffWizard.fromResourceToItem)}
-                refs={refs}
+  if (isError) {
+    return <ErrorView ariaLabel="CompareView-Error" retry={refetch} message={error.message} />;
+  }
+
+  if (isSuccess) {
+    const filteredData = data
```

Comment:
```suggestion
    const filteredResources = data
```
data doesn't bring much meaning, so if we are assigning stuff, we better rename it to something clear. I don't mind when it's directly used from the query itself, but if we are assigning, we should always opt for the better name.